mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/MassIndexingTypeGroupMonitor.java

```diff
@@ -29,22 +27,22 @@ public interface MassIndexingTypeGroupMonitor {
 	 * <p>
 	 * This method can be invoked from several threads thus implementors are required to be thread-safe.
 	 *
-	 * @param increment additional number of documents built
+	 * @param increment The additional number of documents built and added to the index.
 	 */
-	void documentsAdded(long increment);
+	void documentsIndexed(long increment);
 
 	/**
 	 * Notify the monitor that indexing of the type group is starting
 	 * and provide the expected number of entities in the group, if known.
 	 *
-	 * @param totalCount An optional containing the expected number of entities in the group to index, if known,
+	 * @param context An optional containing the expected number of entities in the group to index, if known,
 	 * otherwise an empty optional is supplied.
 	 */
-	void indexingStarted(OptionalLong totalCount);
+	void indexingStarted(MassIndexingTypeGroupMonitorContext context);
 
 	/**
 	 * Notify the monitor that indexing of the type group is completed.
 	 */
-	void indexingCompleted();
+	void indexingCompleted(MassIndexingTypeGroupMonitorContext context);
```

Comment:
I used the same context for both of these, and for now there's only a total count